parlai/core/torch_agent.py

```diff
@@ -712,26 +881,9 @@ def _vectorize_text(self, text, add_start=False, add_end=False,
                               False for targets.
         """
         vec = self.dict.txt2vec(text)
-        if truncate is None or len(vec) + add_start + add_end < truncate:
-            # simple: no truncation
-            if add_start:
-                vec.insert(0, self.START_IDX)
-            if add_end:
-                vec.append(self.END_IDX)
-        elif truncate_left:
-            # don't check add_start, we know are truncating it
-            if add_end:
-                # add the end token first
-                vec.append(self.END_IDX)
-            vec = vec[len(vec) - truncate:]
-        else:
-            # truncate from the right side
-            # don't check add_end, we know we are truncating it
-            vec = vec[:truncate - add_start]
-            if add_start:
-                # always keep the start token if it's there
-                vec.insert(0, self.START_IDX)
-        tensor = torch.LongTensor(vec)
+        vec = self._add_start_end_tokens(vec, add_start, add_end)
```

Comment:
I'm tempted to propose we have `add_start` and `add_end` as arguments to txt2vec instead. It would make memoization a lot easier.

Reply:
like in `DictionaryAgent`?

Reply:
Yeah, not in this PR though.